cmake/FlatccGenerateSources.cmake

```diff
@@ -90,15 +86,28 @@ cmake_minimum_required(VERSION 3.3)
 # TODO: not needed when cmake_minimum_required_version >= 3.4
 include(CMakeParseArguments)
 
-function(flatcc_generate_sources NAME)
```

Comment:
I modelled the function in the idea of `BISON_TARGET` of [`FindBISON`](https://cmake.org/cmake/help/latest/module/FindBISON.html).
Because `NAME` is an argument you need to add always as it needs to be added in the dependency tree,
I opted to have it as a first argument.

See also [`CMakePackageConfigHelpers`](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html): it contains a combination of required positional and keyword arguments.

Reply:
Ok, it is possible that people forget about the first positional argument, but then it will just not work.
People should read the documentation.

Reply:
Yes, people should read documentation, but functions used by end-users should validate input.
I forgot to add <name> in one of the calls in our cmake code and got quite a strange error message.

If you want to keep the <name> argument you can add the following check instead:

```cmake
    #check if caller forgot to provide mandatory argument <name>
    set(ALL_ARGS ${NO_VAL_ARGS} ${SINGLE_VAL_ARGS} ${MULTI_VAL_ARGS})
    list(FIND ALL_ARGS "${NAME}" FLATCC_NAME_INDEX)
    if (NOT FLATCC_NAME_INDEX EQUAL "-1")
        message(FATAL_ERROR "Argument <name> missing in call to flatcc_generate_sources")
    endif()
```